src/textual/events.py

```diff
@@ -411,3 +416,37 @@ class DescendantFocus(Event, verbosity=2, bubble=True):
 
 class DescendantBlur(Event, verbosity=2, bubble=True):
     pass
+
+
+@rich.repr.auto
+class ModeReport(InputEvent):
```

Comment:
This isn't really an InputEvent per se (i.e. key / mouse). It's also not really an event, as it is not relevant to any widgets.

This information is only relevant to the App, so it is probably best as a simple Message sent directly to the App class.